common/vm_wait_guest_hostname.yml

```diff
@@ -29,13 +29,29 @@
   retries: "{{ ((vm_get_hostname_timeout | default(300) | int) / 5) | int }}"
   register: vm_guest_facts
   until:
-    - "'instance' in vm_guest_facts"
+    - vm_guest_facts.instance.guest.toolsRunningStatus | default('') == "guestToolsRunning"
+    - vm_guest_facts.instance.guest.hostName | default('')
+    - (not wait_guest_hostname) or (wait_guest_hostname and wait_guest_hostname == vm_guest_facts.instance.guest.hostName)
+  ignore_errors: true
+
+- name: "Display the retrieved guest info"
+  ansible.builtin.debug: var=vm_guest_facts
+  when: enable_debug | default(false)
```

Comment:
enable_debug has default value.

Reply:
It is convenient for testing the single common task file, so I will leave it there. Thanks~

Reply:
You can just give a value of this 'var' of "vars" in your testing playbook.